src/Pact/Types/PactValue.hs

```diff
@@ -89,3 +107,216 @@ toPactValueLenient t = case toPactValue t of
   Right (PLiteral (LInteger l)) -> PLiteral (LDecimal (fromIntegral l))
   Right v -> v
   Left _ -> PLiteral $ LString $ renderCompactText t
+
+
+
+-- |  Estimate of number of bytes needed to represent data type
+--
+-- Assumptions: GHC, 64-bit machine
+-- General approach:
+--   Memory Consumption = Constructor Header Size + Cost of Constructor Field(s)
+--   Cost of Constructor Field(s)* = 1 word per field + cost of each field's value
+-- (*) See Resource 2 for exceptions to these rules
+-- Resources:
+-- 1. http://wiki.haskell.org/GHC/Memory_Footprint
+-- 2. https://stackoverflow.com/questions/3254758/memory-footprint-of-haskell-data-types
+
+class SizeOf t where
```

Comment:
Let's move this to a `Pact.Types.SizeOf` file with all the instances in there

Reply:
See latest commit.